### PR TITLE
feature/attachments

### PR DESCRIPTION
**Pull request by noxi333**

### CODE REVIEW

File: database/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /data/*
+/storage/*
```
Comment (mvegter) [inline database/.gitignore:2]: newline

File: docker-compose.yml [hunk 2]
```diff
@@ -21,6 +21,11 @@ services:
         read_only: true
         source: ./scripts
         target: /opt
+      - type: bind
+        read_only: false
+        source: ./database/storage
+        target: /var/storage
+
```
Comment (mvegter) [inline docker-compose.yml:28]: One line too many

File: lib/database/repositories/AttachmentRepository.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    /**
+     * Returns all entities queried by log id.
+     *
+     * @param {number} logId the id to query on.
+     * @param {Object} queryBuilder The QueryBuilder to use.
+     * @returns {Promise} Promise object representing the full mock data
+     */
+    async findAllByLogId(logId, queryBuilder = new QueryBuilder()) {
+        queryBuilder.include({
```
Comment (mvegter) [inline lib/database/repositories/AttachmentRepository.js:56]: I think the `whereAssociation` is a better fit for this.

File: lib/database/utilities/QueryBuilder.js [hunk 4]
```diff
@@ -137,6 +137,20 @@ module.exports = (sequelize, Sequelize) => {
...
         }
+
+        /**
+         * Sets a starts with filter using the provided value.
+         *
+         * @param {symbol} value The start value
+         * @returns {QueryBuilder} The current QueryBuilder instance.
+         */
+        startsWith(value) {
+            this.queryBuilder.options.where[this.column] = {
```
Comment (mvegter) [inline lib/database/utilities/QueryBuilder.js:148]: Can you include the not version as well

File: lib/domain/dtos/CreateAttachmentDto.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        .required()
+        .min(3),
+    log: Joi.number()
+        .required(),
+});
+
+const FilesDto = Joi.array().items(FileDto);
+
+const CreateAttachmentDto = Joi.object({
+    query: Joi.object({}),
```
Comment (mvegter) [inline lib/domain/dtos/CreateAttachmentDto.js:28]: Query can contain the token

File: lib/server/utilities/dtoValidator.js [hunk 6]
```diff
@@ -21,11 +21,13 @@
  *                          HTTP request.
  * @returns {Object} Returns the validated value or null if an error occurred.
  */
-const dtoValidator = async (dto, { body, params, query }, response) => {
+const dtoValidator = async (dto, { body, params, query, file = undefined, files = undefined }, response) => {
     let value = null;
+    const request = { body, params, query };
+    Object.assign(request, ...[file, files].filter((f) => f));
```
Comment (mvegter) [inline lib/server/utilities/dtoValidator.js:27]: Shouldn't we use the deepmerge here?

File: lib/usecases/log/CreateLogAttachmentUseCase.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+     * Executes this use case.
+     *
+     * @param {Object} dto The CreateLogDto containing all data.
+     * @returns {Promise}  Promise object represents the result of this use case.
+     */
+    execute(dto) {
+        const { files, params } = dto;
+
+        return TransactionHelper
+            .provide(() => AttachmentRepository.bulkInsert(files.map((file) => ({ ...file, logId: params.logId }))));
```
Comment (mvegter) [inline lib/usecases/log/CreateLogAttachmentUseCase.js:37]: Perhaps store the output in a variable to clarify the resulting array

File: lib/database/repositories/AttachmentRepository.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+            required: true,
+            through: { where: { log_id: logId } },
+        });
+        return this.findAll(queryBuilder);
+    }
+
+    /**
+     * Find one model.
+     *
+     * @param {Object} queryBuilder The Querybuilder to use.
```
Comment (mvegter) [inline lib/database/repositories/AttachmentRepository.js:68]:
```suggestion
     * @param {Querybuilder} queryBuilder The Querybuilder to use.
```

And everywhere else

**Comment by mvegter** [comment]
Reminder to execute:
https://github.com/AliceO2Group/Bookkeeping/blob/master/docs/DEVELOPMENT.md#updating-transitive-dependencies

**Comment by mvegter** [comment]
please rebase

File: lib/usecases/log/CreateLogAttachmentUseCase.js (hunk 7 above)
Comment (noxi333) [reply lib/usecases/log/CreateLogAttachmentUseCase.js:37]: The output of the bulkInsert or the output of the map?

**Pull request merged**